src/components/core/PivotTable.tsx

```diff
@@ -428,18 +430,27 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             execution: null,
             columnTotals: cloneDeep(this.getColumnTotalsFromResultSpec(this.props.resultSpec)),
             agGridRerenderNumber: 1,
-            desiredHeight: props.config.maxHeight
+            desiredHeight: props.config.maxHeight,
+
+            useGrouping: true
         };
 
         this.agGridDataSource = null;
         this.gridApi = null;
-        this.groupingProvider = GroupingProviderFactory.createProvider(props.groupRows);
+
+        this.setGroupingProvider(props.groupRows);
     }
 
     public componentWillMount() {
         this.createAGGridDataSource();
     }
 
+    public componentWillUpdate(nextProps: IPivotTableInnerProps, nextState: IPivotTableState) {
+        if (this.props.groupRows !== nextProps.groupRows || this.state.useGrouping !== nextState.useGrouping) {
```

Comment:
`useGrouping` instead of this consider for example name `isSortedFirstAttribute`